File: algorithms/2_medium/C/01006_clumsy_factorial/simulation.py
class Solution:
    def clumsy(self, n: int) -> int:
        is_first = True
        res = 0

        while n > 0:
            curr = n
            n -= 1

            if n > 0:
                curr *= n
                n -= 1

            if n > 0:
                curr //= n
                n -= 1

            if is_first:
                res += curr
                is_first = False
            else:
                res -= curr

            if n > 0:
                res += n
                n -= 1

        return res

File: algorithms/2_medium/C/01006_clumsy_factorial/simulation.cpp
class Solution {
public:
    int clumsy(int n) {
        bool is_first{ true };
        int res{ 0 };

        while (n > 0) {
            int curr{ n };
            n -= 1;

            if (n > 0) {
                curr *= n;
                n -= 1;
            }

            if (n > 0) {
                curr /= n;
                n -= 1;
            }

            if (is_first) {
                res += curr;
                is_first = false;
            } else {
                res -= curr;
            }

            if (n > 0) {
                res += n;
                n -= 1;
            }
        }

        return res;
    }
};
